backends/memcache.go

```diff
@@ -67,44 +69,108 @@ func (m *Memcache) Execute(cmd string, args ...interface{}) (interface{}, error)
 	return "inside GEt", nil
 }
 
-func (m *Memcache) Delete(keys ...interface{}) int {
-	return 1
+func (m *Memcache) Delete(keys ...interface{}) (int, error) {
```

Comment:
My view was client should know what error has occurred from datastore side, so rather than keeping called client uninformed, should return error as well.

This is anyways golang convention to return (resp, error)

Don't know about separate channel here; would be unnecessary overhead.